compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1326,11 +1331,12 @@ class RefChecks extends MiniPhase { thisPhase =>
 
   override def transformIdent(tree: Ident)(using Context): Tree =
     checkAnyRefMethodCall(tree)
+    checkSynchronizedCallOnValue(tree)
     tree
 
   override def transformSelect(tree: tpd.Select)(using Context): tpd.Tree =
     if defn.ScalaBoxedClasses().contains(tree.qualifier.tpe.typeSymbol) && tree.name == nme.synchronized_ then
-      report.warning(SynchronizedCallOnBoxedClass(tree), tree.srcPos)
+      report.warning(SynchronizedCallOnValue(tree), tree.srcPos)
```

Comment:
This should probably reuse `checkSynchronizedCallOnValue`. Otherwise explicit selections on `AnyVal` won't do the right thing, AFAICT.

Reply:
We tried the following during the Spree:

```scala
//> using options -explain
class A(val s: String) extends AnyVal
def Test =
  val a = A("hello")
  a.synchronized { println("hello, world") } // warn
```

And this is already throws an error:

```
 ~/scala-snippets-5 scalac foo.scala
-- [E008] Not Found Error: foo.scala:5:4 ---------------------------------------
5 |  a.synchronized { println("hello, world") } // warn
  |  ^^^^^^^^^^^^^^
  |value synchronized is not a member of A, but could be made available as an extension method.
  |
  |The following import might fix the problem:
  |
  |  import scala.reflect.Selectable.reflectiveSelectable
  |
1 error found
```

Is this the case you have in mind?

Anyway, I guess  it might be nice to use the same logic both for `Select` and `Apply`.